[Refactor] PawcutCameraView 리펙토링

hkwangro opened the pull request:
## 변경 사항 (Changes)

- PawcutCameraView 변수로 불러오는 부분을 Pawcut 스타일 가이드 맞게 변경 했습니다.


<img width="344" height="163" alt="image" src="https://github.com/user-attachments/assets/3c04dfbb-43cc-4e78-9f22-4db42eceba9c" />

- 위 이미지 처럼  1) PawcutCameraBottomView  2) PawcutCameraTopView 를 추가하여 가독성을 높였습니다.

```
var body: some View {
        ZStack {
            backgroundLayer
            cameraPreviewLayer
            shutterAnimationLayer
            uiOverlayLayer
        }
        private var uiOverlayLayer: some View {
        ZStack {
            topNavigationView
            shotProgressView
            countdownView
            bottomControlsView
            tooltipView
        }
    }
```

위 코드 처럼 변수로 불러 오는 부분을 ZTackt 안에 정리했고,
VStack 부분을 위 이미지와 같이 SubView로 옮겨서 리펙토링 했습니다.


## 변경 이유 (Reason for Changes)
## 관련 이슈 (Related Issue)
closed #180 

## 테스트 방법 (How to Test)

## 추가 정보 (Additional Information)
```
 if let timeLeft = viewModel.countdownNumber {
                    Text("\(timeLeft)")
                    .font(.system(size: 100, weight: .bold))
                    .foregroundColor(.white)
                    .shadow(radius: 5)
                    .transition(.opacity)
                    .padding(.bottom, 60)
            }
```

위의 57~65 번줄은 컴포넌트화 되어 있지 않아서, 기존 코드를 유지 했습니다.

hkwangro committed:
refactor: 변수명 정리하고 일차 정리

변수명 정리하고 일차 정리

hkwangro committed:
refactor: ZStack 개선

-Group 삭제
-스타일 가이드 맞춤

hkwangro committed:
refactor: PawcutCameraView 정리

- PawcutCameraBottomSubView
- PawcutCameraTopSubView

나눠서 정리

hkwangro committed:
fix: Text

- 타이머 Text 여서 컴포넌트 적용이 되지 않아 원래 코드로 변경

luminouxx commented on PawCut/PawCut/Sources/Views/PawcutCamera/SubViews/PawcutCameraBottomSubView.swift:
```diff
@@ -0,0 +1,91 @@
+import SwiftUI
+
+struct PawcutCameraBottomSubView: View {
+    @ObservedObject var viewModel: PawcutCameraViewModel
```
하위 뷰에 ViewModel 말고 메서드를 전달하는 방식으로 수정해주세요!

luminouxx commented on PawCut/PawCut/Sources/Views/PawcutCamera/PawcutCameraView.swift:
```diff
@@ -56,219 +27,53 @@ struct PawcutCameraView: View {
+                .offset(y: -270)
+            
             if let timeLeft = viewModel.countdownNumber {
-                Text("\(timeLeft)")
+                    Text("\(timeLeft)")
                     .font(.system(size: 100, weight: .bold))
                     .foregroundColor(.white)
                     .shadow(radius: 5)
                     .transition(.opacity)
                     .padding(.bottom, 60)
-            }
```
이 부분을 Optional Binding을 하기 보다는 Nil 병합 연산자 (??)를 활용하면 가독성이 조금 더 좋을 것 같습니다.
```swift
Text("\(viewModel.countdownNumber ?? 0)")
      .font(.system(size: 100, weight: .bold))
      .foregroundColor(.white)
      .shadow(radius: 5)
      .transition(.opacity)
      .padding(.bottom, 60)
```

hkwangro replied:
오 감사합니다!!!

luminouxx commented on PawCut/PawCut/Sources/Views/PawcutCamera/PawcutCameraView.swift:
```diff
@@ -56,219 +27,53 @@ struct PawcutCameraView: View {
                         height: UIScreen.main.bounds.width * 4 / 3
                     )
                     .offset(y: -40)
             }
```
이 부분도 if let (옵셔널 바인딩) 보다는 Nil 병합 연산자를 활용하는게 가독성 및 코드 길이상 이점이 있을 것 같아요.

hkwangro replied:
넵넵 수정 했습니다!!

luminouxx reviewed (changes requested):
광로 고생 많으셨습니다!

1. Optional Binding 보다는 Nil 병합 연산자를 UI 코드에서 활용하는 것으로 해주세요!
2. Subview(하위 컴포넌트에) ViewModel을 전달하고 있는데, 필요한 Method만 파라미터로 넘겨주세요. (버튼 컴포넌트 썼던 것 기억하시죠?)

hkwangro committed:
fix: 수정내용 업데이트

- Optional Binding 보다는 Nil 병합 연산자를 UI 코드에서 활용하는 것으로 변경
- Subview(하위 컴포넌트에) ViewModel을 전달하고 있는데, 필요한 Method만 파라미터로 변경
